src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -22,19 +52,24 @@ export const TodoFilter: React.FC = () => {
           type="text"
           className="input"
           placeholder="Search..."
+          value={localQuery}
+          onChange={handleInput}
         />
         <span className="icon is-left">
           <i className="fas fa-magnifying-glass" />
         </span>
 
-        <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-          {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-          <button
-            data-cy="clearSearchButton"
-            type="button"
-            className="delete"
-          />
-        </span>
+        {query.length > 0 && (
```

Comment:
The `query` variable is used to determine if the clear button should be shown, but it is not updated when `localQuery` changes. If `query` is supposed to reflect the current input state, ensure it is synchronized with `localQuery`.